### PR TITLE
V360: Denial reasons not equal

### LINKED ISSUE #163: Validation (2017): V360 (Reasons for Denial)
**Edit #:** V360
**Type:** Validity
**Category:** Loan/Application Register (only)
**Column:** Reasons for Denial

Responses for reasons for denial must not be the same (i.e. 1,1; 1,1,2; 1,2,2;...........).

Additional resources:
* [FFIEC's 2016 HMDA File Spec](https://www.ffiec.gov/hmda/pdf/spec2016.pdf)
* [FFIEC's 2016 HMDA Edits](https://www.ffiec.gov/hmda/pdf/edit2016.pdf)
* [CFPB's 2017 HMDA File Spec](http://www.consumerfinance.gov/hmda/static/for-filers/2017/2017-HMDA-File-Specifications.pdf)

### PR DESCRIPTION
**Pull request by nickgrippin**
Closes #163

### CODE REVIEW

File: validation/src/main/scala/hmda/validation/dsl/PredicateCommon.scala [hunk 1]
```diff
@@ -54,7 +54,8 @@ object PredicateCommon {
   def numeric[T]: Predicate[T] = new Predicate[T] {
     override def validate: (T) => Boolean = _.asInstanceOf[AnyRef] match {
       case n: Number => true
-      case _ => throw new NotImplementedError("'numeric' doesn't handle string (or other) values yet")
+      case s: String => (s forall Character.isDigit) && !s.isEmpty
```
Comment (schbetsy) [inline validation/src/main/scala/hmda/validation/dsl/PredicateCommon.scala]: Looks like this case (`case s: String => ...`) is no longer necessary

Comment (nickgrippin) [inline validation/src/main/scala/hmda/validation/dsl/PredicateCommon.scala]: True, but I feel like this code could be helpful in the future.  Would it be better to make this change as a separate, one-line pull request (which might satisfy codecov)?

Comment (schbetsy) [inline validation/src/main/scala/hmda/validation/dsl/PredicateCommon.scala]: Sure. Then we can discuss it without blocking this PR.

File: validation/src/test/scala/hmda/validation/rules/lar/validity/V360Spec.scala [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+import hmda.model.fi.lar.{ Denial, LoanApplicationRegister }
+import hmda.validation.dsl.{ Failure, Success }
+import hmda.validation.rules.EditCheck
+import hmda.validation.rules.lar.LarEditCheckSpec
+import org.scalacheck.Gen
+
+class V360Spec extends LarEditCheckSpec {
+  property("Denials with all different values must pass") {
+    forAll(larGen) { lar =>
+      val goodDenial = lar.denial.copy(reason1 = "1", reason2 = "2", reason3 = "")
```
Comment (schbetsy) [inline validation/src/test/scala/hmda/validation/rules/lar/validity/V360Spec.scala]: I'd like to see this test be a bit more generic. For instance, you could use a generator like `badOptionGen` below to generate three values, then just use a `whenever` clause to make sure they're all different.

**Comment by schbetsy** [comment]
LGTM

**Pull request merged**